### PR TITLE
Improve flushing multiple column families

### PR DESCRIPTION
**Pull request by riversand963**
If one column family is dropped, we should simply skip it and continue to flush
other active ones.
Currently we use Status::ShutdownInProgress to notify caller of column families
being dropped. In the future, we should consider using a different Status code.

Test plan:
```
$make clean && make -j32 all check
$./db_stress -atomic_flush -reopen=0 -disable_wal=1
```

### CODE REVIEW

**Comment by riversand963** [comment]
Could you PTAL? @ajkr

File: db/db_impl_compaction_flush.cc [hunk 1]
```diff
@@ -341,28 +345,34 @@ Status DBImpl::AtomicFlushMemTablesToOutputFiles(
...
-        s = exec_status[0].second;
+    exec_status[0].second =
+        jobs[0].Run(&logs_with_prep_tracker_, &file_meta[0]);
+    exec_status[0].first = true;
+  }
+
+  Status error_status;
+  for (const auto& e : exec_status) {
+    if (!e.second.ok()) {
+      s = e.second;
```
Comment (ajkr) [inline db/db_impl_compaction_flush.cc]: what if `s` has a worse status than `ShutdownInProgress` prior to reaching this point and then here we overwrite it with `ShutdownInProgress`?

File: db/version_set.cc [hunk 2]
```diff
@@ -2857,7 +2857,32 @@ Status VersionSet::ProcessManifestWrites(
       last_writer = *(it++);
       assert(last_writer != nullptr);
       assert(last_writer->cfd != nullptr);
-      if (last_writer->cfd != nullptr && last_writer->cfd->IsDropped()) {
+      if (last_writer->cfd->IsDropped()) {
```
Comment (ajkr) [inline db/version_set.cc]: is it possible to simply recompute all the `VersionEdit::remaining_entries_`s here? I find the fixup logic kind of hard to follow.

File: db/db_impl_compaction_flush.cc (hunk 1 above)
Comment (riversand963) [reply db/db_impl_compaction_flush.cc]: You are right, will fix it.

File: db/version_set.cc (hunk 2 above)
Comment (riversand963) [reply db/version_set.cc]: This approach performs recomputation only when some column family has been dropped. Are you suggesting that we recompute this information anyway even if no column family is dropped?

Comment (ajkr) [reply db/version_set.cc]: If it simplifies the code and doesn't increase the complexity, I'd vote for it.

Comment (riversand963) [reply db/version_set.cc]: After some investigation, I still prefer the current approach because recomputing does not seem to simplify the code.

Comment (ajkr) [reply db/version_set.cc]: Sounds good, thanks for looking into it!

**Comment by riversand963** [comment]
Added some unit tests for dropped column family handling.
